### PR TITLE
feat(app): update device reset for gripper calibration

### PR DESCRIPTION
**Pull request by koji**
# Overview
update device reset component to activate gripper calibration reset and update buttons' styles.
In terms of `Factory reset`, it will be implemented by a following PR and it has been ticketed already.

design
https://www.figma.com/file/AoTLAYuWawlaWItB1umOjr/Release%3A-Opentrons-Flex-Touchscreen?type=design&node-id=7948-118690&mode=design&t=AwB5o6W0Bsxv2Gmy-0

Screenshot 2023-07-05 at 3 11 45 PM

There are a couple of things I need to ask about design to the design team.
- Desktop app uses `Factory Reset` but Touch screen uses `Factory reset`
- In Touchscreen design uses `Pipette Offset Calibrations` and `Pipette Offset Calibration(s)`

# Test Plan

# Changelog
- update DeviceReset components and tests

# Review requests

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx [hunk 1]
```diff
@@ -78,15 +80,23 @@ export function DeviceReset({
     }
   }
 
+  let subText: string
   const renderText = (optionId: string): string => {
```
Comment (shlokamin) [inline app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: rather than having the `renderText` function mutate a global variable (`subText`), can we rename + update this function to return text AND subtext? then we can use it when we render JSX.

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx [hunk 2]
```diff
@@ -78,15 +80,23 @@ export function DeviceReset({
...
         return t('clear_option_pipette_calibrations')
-      // ToDo (kj:02/07/2023) gripperCalibration same as the above
-      case 'gripperCalibration':
+      case 'gripperOffsetCalibrations':
         return t('clear_option_gripper_calibration')
       case 'runsHistory':
+        subText = t('clear_option_runs_history_subtext')
         return t('clear_option_runs_history')
+      // ToDo (kj:07/01/2023) we need to add factory reset case
+      // Asking a question about text Desktop Factory Reset/Touchscreen Factory rest
```
Comment (shlokamin) [inline app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: what's the question? lmk how i can help unblock you!

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx [hunk 3]
```diff
@@ -128,33 +138,37 @@ export function DeviceReset({
...
+                  </StyledText>
+                  {option.id === 'runsHistory' ? (
+                    <StyledText
+                      as="p"
+                      color={
+                        resetOptions[option.id]
+                          ? COLORS.white
+                          : COLORS.darkBlack70
+                      }
+                    >
```
Comment (shlokamin) [inline app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: i don't follow this block — why is this necessary?

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx (hunk 2 above)
Comment (koji) [reply app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: Actually this part was solved on Monday.
I will remove the comment

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx (hunk 3 above)
Comment (koji) [reply app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: A couple of options have subtext and it is needed to change the text color when a option is selected or deselected

Comment (shlokamin) [reply app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: but this is only for runs history, why is that one special? 

```js
{option.id === 'runsHistory' ? (
```

Comment (koji) [reply app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: Because when I opened this PR, runHistory was the only one item that has subText.
But I just updated this part.

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx [hunk 4]
```diff
@@ -40,8 +40,10 @@ const OptionLabel = styled.label<LabelProps>`
     ${({ isSelected }) =>
       isSelected === true ? COLORS.blueEnabled : COLORS.light2};
```
Comment (brenthagen) [inline app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: i don't see any border on this button in figma

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx [hunk 5]
```diff
@@ -111,50 +134,60 @@ export function DeviceReset({
...
         </Flex>
-        <AlertPrimaryButton
-          paddingY={SPACING.spacing24}
+        <MediumButton
+          data-testid="DeviceReset_clear_data_button"
+          buttonText={t('clear_data_and_restart_robot')}
+          buttonType="alert"
           disabled={
             Object.keys(resetOptions).length === 0 ||
             availableOptions.every(option => resetOptions[option.id] === false)
```
Comment (brenthagen) [inline app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: this button remains enabled after selecting and "unselecting" an option - we need to check for whether the option exists in `resetOptions` in addition to its value:
```
            availableOptions.every(
              option =>
                resetOptions[option.id] === false ||
                resetOptions[option.id] === undefined
            )
```

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx (hunk 4 above)
Comment (koji) [reply app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: oh good catch, I forgot to remove when the design was update.

Comment (jerader) [inline app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: Looks like Brent commented on there not needing a border but i still see a border? 
```suggestion
      isSelected === true ? COLORS.blueEnabled : undefined};
```
or
```suggestion
      isSelected === true ? COLORS.blueEnabled : COLORS.mediumBlueEnabled};
```

**Review by jerader — APPROVED** [review]
lgtm but left a question about border

**Review by brenthagen — APPROVED** [review]

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx (hunk 4 above)
Comment (jerader) [reply app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: on nevermind, i must have been looking at an older commit!

Comment (jerader) [reply app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]: looks good now 😄
